src/inventory.cc

```diff
@@ -2366,11 +2355,12 @@ static void _inven_pickup(int buttonCode, int indexOffset)
         _inven_update_lighting(nullptr);
     }
 
-    // SFALL: allow ctrl-click to unequip
-    bool immediate = itemIndex == -1 && _ctrl_pressed();
+    // allow ctrl-click to quick unequip or equip item
+    bool immediate = _ctrl_pressed();
```

Comment:
Everything above this is cleanup only (replacing magic numbers with constants and the like).  Below this is the implementation